server.js

```diff
@@ -437,8 +439,8 @@ exports.buildExpress = function(options) {
           attachments.push(fileMeta);
         }
       });
-      attachments.concat(demo.attachments);
-      demo.attachments = attachments;
+      var oldAttachments = replacer(attachments, demo.attachments);
```

Comment:
The use of replacer looks a bit odd. It took me a while to understand what it was doing, but apparently demo data is passed in as string, and not as json. I would have expected the json bodyparser to deal with that, but apparently not?

Anyhow, I think just replacing attachments is not enough, I think you need to json parse the entire demo. I had to use the following to make sure demos were saved correctly. Maybe you can review, I wouldn't be surprised it needs to be applied in multiple places, or maybe there is a certain reason why the bodyparser isn't doing anything?

I replaced:
    var demo = _.extend({},req.body);
with:
    var demo = _.extend({},JSON.parse(req.body.data));

Or could this just be a side-effect of using $upload for uploading attachments?